core/src/test/java/com/linecorp/armeria/server/ServiceRequestCancellationTest.java

```diff
@@ -57,14 +60,16 @@ protected void configure(ServerBuilder sb) {
     @EnumSource(value = SessionProtocol.class, names = {"H1C", "H2C"})
     @ParameterizedTest
     void shouldCompleteLogWhenCancelledByClient(SessionProtocol protocol) {
-        final ClientFactory factory = ClientFactory.builder().build();
         final WebClient client = WebClient.builder(server.uri(protocol))
-                                          .factory(factory)
                                           .build();
-
-        final CompletableFuture<AggregatedHttpResponse> responseFuture = client.get("/reset").aggregate();
+        final CompletableFuture<AggregatedHttpResponse> responseFuture;
+        final ClientRequestContext clientRequestContext;
+        try (ClientRequestContextCaptor captor = Clients.newContextCaptor()) {
+            responseFuture = client.get("/reset").aggregate();
+            clientRequestContext = captor.get();
+        }
         await().untilAtomic(ctxRef, Matchers.notNullValue());
-        factory.close();
```

Comment:
Previously, it closes the stream right away without waiting for the graceful shutdown.
But now, it's changed to close the connection after graceful shutdown so we cannot use `factory.close()` for the test. So I used ctx.cancel instead.